callgraph/core.py

```diff
@@ -52,17 +55,19 @@ def __init__(self, name):
     def AddConnection(self, dst, kind, line_number=NO_LINE_NUMBER):
         self.connections.add(Connection(dst, kind, line_number))
 
-    def UpdateWidth(self):
-        self.node_width = self.loc * NODE_WIDTH_FACTOR
+    def UpdateSize(self):
+        self.node_width = int(round(self.loc * NODE_WIDTH_FACTOR))
+        
+        if self.node_width > MAX_NODE_WIDTH:
+            self.node_width = MAX_NODE_WIDTH
+
+        self.node_height = self.node_width // 2

```

Comment:
Imagine that you have nodes varying in size between 1 and 400 lines of code. In this case, node size will increase linearly from 1 to 5, but all nodes with size 5 and below will have size 5, thus making it impossible to distinguish them in the output.

Here is an example node size distribution from a script I found around:

andrea@SPADACCIO-DESK:/mnt/c/sd$ egrep -o "[0-9]* LOC" example.dot | sort -n                                                                                                                4 LOC
6 LOC
6 LOC
8 LOC
8 LOC
9 LOC
9 LOC
9 LOC
9 LOC
9 LOC
9 LOC
11 LOC
11 LOC
12 LOC
12 LOC
12 LOC
13 LOC
14 LOC
14 LOC
15 LOC
15 LOC
16 LOC
17 LOC
17 LOC
18 LOC
19 LOC
19 LOC
20 LOC
21 LOC
23 LOC
23 LOC
30 LOC
36 LOC
43 LOC
51 LOC
53 LOC
57 LOC
93 LOC
96 LOC
118 LOC
150 LOC
357 LOC

All these nodes except one would in this script have width 5 and height 2.5.

What I would suggest is to specify a larger max size (ideally overridable as a command-line parameter), use the max size as the size for the largest node and then have the others have a size proportional to the number of LOC in the scale between a given minimum width (maybe 3?) and the max size.

Makes sense?

Reply:
Gotcha, let me do it.

Reply:
Is it possible to have more .cmd files as examples?

Reply:
The tool is here if you want to play with it a bit more: https://github.com/microsoft/cmd-call-graph/blob/master/scripts/generate-sample-cmd.py

Reply:
Thank you. I am going to use it to generate more sample .cmd files. Thank you.